### ISSUE TITLE
Bug: slf4j-log4j12 dependency scope

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [moquette-io/moquette#408]
1. normally, `slf4j-api` implementations are provided by user, not library;
currently `slf4j-log4j12` is in dependency `scope=compile`, but should be `test`

https://github.com/andsel/moquette/blob/master/broker/build.gradle#L30
```groovy
compile group: 'org.slf4j', name: 'slf4j-log4j12', version:'1.7.5'
```
2. one work around is to provide explicit dependency exclusion:
```xml
		<dependency>
			<groupId>io.moquette</groupId>
			<artifactId>moquette-broker</artifactId>
			<version>0.11</version>
			<exclusions>
				<exclusion>
					<groupId>org.slf4j</groupId>
					<artifactId>slf4j-log4j12</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
```

**Comment by andsel** [comment]
Hi @Andrei-Pozolotin thanks, solved with e824a22ae61ec739eec52ede71e553901321baef fixed the broker lib to depends only on the API while the dep on Impl has been moved to the distribution

**Comment by Andrei-Pozolotin** [comment]
thank you
